model/EligibilityService.php

```diff
@@ -94,12 +95,35 @@ public function getAssignmentService() {
      * @param Resource $testCenter
      * @param Resource $delivery
      * @return boolean
+     *
+     * @deprecated use EligibilityService::newEligibility()
      */
     public function createEligibility(Resource $testCenter, Resource $delivery) {
-        if (!is_null($this->getEligibility($testCenter, $delivery))) {
-            // already exists, don't recreate
+
+        try {
+            return (boolean) $this->newEligibility($testCenter, $delivery);
+        } catch (DuplicateResourceException $e) {
             return false;
         }
+    }
+
+    /**
+     * Establishes a new eligibility
+     *
+     * @param Resource $testCenter
+     * @param Resource $delivery
+     * @return Resource
+     *
+     * @throws DuplicateResourceException
+     * @throws \common_exception_InconsistentData
+     * @throws \core_kernel_persistence_Exception
+     */
+    public function newEligibility(Resource $testCenter, Resource $delivery)
+    {
+        if (!is_null($this->getEligibility($testCenter, $delivery))) {
```

Comment:
A really-really small performance improvement could be if you replace `!is_null(...)` to `... !== null` :)